card.go

```diff
@@ -170,3 +185,24 @@ func deepClone(ix interface{}) interface{} {
 	}
 	return ix
 }
+
+func fillNameComponentMap(m map[string]interface{}, items interface{}) {
+	for _, v := range items.([]interface{}) {
```

Comment:
Is it always expected to get an array? This will panic if not. It would be safer to expect `items []interface{}` but I assume the variable passed inside (`c.D`) is of type `interface{}`.

Reply:
>Is it always expected to get an array? This will panic if not

Yes, we are only interested in Wave component arrays - `items | secondary_items | buttons` which are all supposed to be of type `Component[]`. If any of them is not, the panic is desired as it hints that something is off with the typing.

I could do an if check for type casting and throw a nicer err than the panic one, but not sure if worth it.

>It would be safer to expect items []interface{} but I assume the variable passed inside (c.D) is of type interface{}

Correct. The casting would need to happen either in outer function or here.

Reply:
> I could do an if check for type casting and throw a nicer err than the panic one, but not sure if worth it.

I'd probably skip it. But maybe a comment would be nice.